lib/bap_llvm/llvm_macho_loader.hpp

```diff
@@ -152,6 +137,38 @@ MACHO_SECTION_FIELD(reserved2, uint32_t)
 MACHO_SECTION_FIELD(addr, uint64_t)
 MACHO_SECTION_FIELD(offset, uint32_t)
 
+void entry_point(const macho &obj, ogre_doc &s) {
+    for (auto info : macho_commands(obj)) {
+        if (info.C.cmd == MachO::LoadCommandType::LC_MAIN) {
+            const MachO::entry_point_command *entry_cmd =
+                reinterpret_cast<const MachO::entry_point_command*>(info.Ptr);
+            uint64_t entry = entry_cmd->entryoff;
+            if (obj.isLittleEndian() != sys::IsLittleEndianHost)
+                sys::swapByteOrder(entry);
+            s.entry("entry") << entry;
+            return;
+        }
+    }
+    std::vector<uint64_t> addrs;
+    for (auto sec : prim::sections(obj)) {
+        auto addr = prim::section_address(sec);
+        if (!addr) continue;
+        if (section_flags(obj, sec) & MachO::S_ATTR_PURE_INSTRUCTIONS)
+            addrs.push_back(*addr);
+    }
+    std::sort(addrs.begin(), addrs.end());
+    if (addrs.size() > 0)
+        s.entry("entry") << addrs.front();
+}
+
+void image_info(const macho &obj, ogre_doc &s) {
+    if (!is_exec(obj))
+        s.raw_entry("(entry 0)");
```

Comment:
why is it 0, but not the lowest address? If it is a library then will this take into account the rebasing?

Reply:
it equals to zero because we use relative addresses. so after rebasing entry point will be equal to the lowest address